### ISSUE TITLE
[Bug]: Static XCFramework assets are embedded in app during compilation on Xcode 15.3 RC (Xcode bug)

### ISSUE THREAD

**Issue by tiwoc** [braze-inc/braze-swift-sdk#105]
### Platform

iOS

### Platform Version

iOS 17.4

### Braze SDK Version

8.0.0

### Xcode Version

Xcode 15.3 beta 3 (15E5194e)

### Computer Processor

Apple (M1)

### Repro Rate

100% of the time

### Steps To Reproduce

1. Use Xcode 15.3 beta 3 to archive a build for _Any iOS device (arm64)_
2. In Organizer > Archives, select the new archive and click _Validate App_ (also happens when trying to upload to the App Store, but validations triggers the bug without even starting the distribution process)
3. Select _Validate_ and click the _Validate_ button

### Expected Behavior

Validation should succeed

### Actual Incorrect Behavior

A validation error is shown:
> <IDEDistributionIssue: severity(error), error(Error Domain=ContentDelivery Code=90208 \"Asset validation failed\" UserInfo={IDEDistributionIssueSeverity=3, NSLocalizedDescription=Asset validation failed, NSLocalizedRecoverySuggestion=Invalid Bundle. The bundle redacted.app/Frameworks/BrazeKit.framework does not support the minimum OS Version specified in the Info.plist. (ID: 6ca384f5-cc0f-40c3-902b-9d1fc26a45ba)})>

This prevents the distribution of the app via TestFlight and the App Store when built with Xcode 15.3.

### Additional Information

I was able to track down the reason for the failure thanks to [a thread in Apple Developer Forums](https://forums.developer.apple.com/forums/thread/705297). There's an inconsistency in the minimum supported iOS version of the BrazeKit framework:
- BrazeKit.framework/Info.plist in the archived app bundle (extracted with `plutil -p Info.plist`):
  ```xml
  …
  <key>MinimumOSVersion</key>
  <string>12.0</string>
  …
  ```
- BrazeKit.framework/BrazeKit in the archived app bundle (extracted with `vtool -show BrazeKit`):
  ```
  BrazeKit:
  Load command 8
        cmd LC_BUILD_VERSION
    cmdsize 32
   platform IOS
      minos 14.0
        sdk 17.4
     ntools 1
       tool LD
    version 1053.12
  Load command 9
        cmd LC_SOURCE_VERSION
    cmdsize 16
    version 0.0
  ```

`MinimumOSVersion` is 12.0, but `minos` is 14.0.

I was able to fix validation locally by bumping `MinimumOSVersion` to 14.0 in `BrazeKit.framework/Info.plist`.

This might be a bug that is introduced or at least triggered by Xcode 15.3. iOS 14.0 is the deployment target of the app that we're including BrazeKit in. If I set the app's deployment target to another iOS version and archive the app again, BrazeKit's `minos` changes to that new version. Xcode 15.3 seems to update Braze's minos but not MinimumOSVersion to the one of the app target.

Interestingly, when archiving with Xcode 15.2, I don't even see BrazeKit.framework in the archived app bundle. I wonder if Xcode 15.2 statically links BrazeKit while Xcode 15.3 does dynamic linking?

**Comment by tiwoc** [comment]
Not sure if it's relevant, but we're including Braze via SPM.

**Comment by tiwoc** [comment]
Interestingly, BrazeKit is the only framework in our app that gets its `minos` bumped by Xcode 15.3. All the others retain their version, so there must be _something_ inherent in BrazeKit, even if the source of the bug is in Xcode.

**Comment by jacksonemiller** [comment]
Hi @tiwoc,

Thank you very much for bringing this to our attention. We've begun looking into this and will let you know of any updates.

Thanks!

**Comment by jacksonemiller** [comment]
Hey, one other thing I forgot to mention. For right now, another potential workaround you might try is using [our pre-built dynamic framework repository](https://github.com/braze-inc/braze-swift-sdk-prebuilt-dynamic).  It seems like it may not have the same issue.

**Comment by tiwoc** [comment]
Hi @jacksonemiller, thanks for bringing this up. With the dynamic framework, we don't run into the validation error 👍 Thank you!

For completeness, I also tried the [braze-swift-sdk-prebuilt-static](https://github.com/braze-inc/braze-swift-sdk-prebuilt-static), but it doesn't change anything, which makes sense if you consider that the base repo also includes a static framework for BrazeKit.

**Comment by kivra-janko** [comment]
Xcode Cloud has just upgraded their "latest release" definition to Xcode 15.3 RC, which means all our builds have started to fail because of this issue.

Ironically, https://developer.apple.com/download/applications/ still lists 15.3 beta 3 as the latest...

**Comment by tiwoc** [comment]
Thanks for the update! The prebuilt dynamic XCFramework works great for us, for the time being.

**Comment by gooson** [comment]
+1

**Comment by hokstuff** [comment]
Hi @tiwoc @kivra-janko @gooson,

We have just released [Braze Swift SDK version 8.2.1](https://github.com/braze-inc/braze-swift-sdk/releases/tag/8.2.1) which fixes the issues during archiving on Xcode 15.3.

Let us know if you run into any further issues - Thanks!

**Comment by tiwoc** [comment]
Works great for us. Thanks, @hokstuff!
